keras/applications/applications_load_weight_test.py

```diff
@@ -115,7 +125,7 @@ def test_application_pretrained_weights_loading(self):
       self.assertShapeEqual(model.output_shape, (None, _IMAGENET_CLASSES))
       x = _get_elephant(model.input_shape[1:3])
       x = app_module.preprocess_input(x)
-      preds = model.predict(x)
+      preds = model(x).numpy()
```

Comment:
Is there a reason this needs to be updated? Do things break otherwise? We still run these test cases in TF1 without eager mode enabled, and this line is breaking a number of our tests.

Reply:
@mattdangerw 

Yes, the change is necessary. Grouped convolutions are not yet fully supported on CPUs. We see that `model.predict(X_test)` breaks whereas `model(X_test)` works fine. 

There are a [number of issues](https://github.com/tensorflow/tensorflow/search?q=grouped+convolution&type=issues) discussing this in the TF repo.

Reply:
I think you could trigger the failures by adding a call to `tf.compat.v1.disable_v2_behavior()` after before the call to `tf.test.main` in `applications_load_weight_test`. We can't submit this if we are breaking all these application tests in a TF1 context. We would need to find a change that does not rely on eager mode behavior (.numpy is eager only).

This might mean we need to dig into the difference between direct call vs predict here. It sound like this is an issue with grouped convolutions on CPU that will only appear when compiling a tf.function, is that right?

Reply:
@mattdangerw 

I have found a small solution to this. I have tested the change both with TF2 and using `tf.compat.v1.disable_v2_behavior()` and it works on my end. Could you please take a look and run the workflow again?

https://github.com/keras-team/keras/blob/851ca160f7c38e9effd9e60bcd44b6611073ae60/keras/applications/applications_load_weight_test.py#L127-L132

Reply:
I will take a look next week! Sorry for the delayed reply, most of the team is out this week. The proposed change would still not submit, because that fallback (the numpy call) would still be run in a TF1 context for the regnet load weights test unless we disable it.

Overall, I think our options are...

 1) Disable the load weights test for regnet (without removing the predict call here), and follow up with a fix.
 2) Fix the underlying CPU/compiled function/grouped convolution issue, and then land this PR.
 3) Work around the bug for regnets somehow (the conversation [here](https://github.com/tensorflow/tensorflow/issues/51825) suggests that using `jit_compile=True` may allow CPU to work, which might give us a way forward).

I would say 3) would be the way to go if we can make it work. We really do want the load weights tests to test the compile predicted function (that's how these will often be used!), and shipping regnets such that predict will fail on CPU by default is not a great out of box experience.

Will follow up next week when people are back in office. Thanks!

Reply:
@mattdangerw 

Thanks for the detailed explanation. 

I guess (2) is not something which can be done in the Keras codebase, as error is thrown in [tensorflow/tensorflow/core/kernels/conv_ops_fused_impl.h](https://github.com/tensorflow/tensorflow/blob/1ef2bbae5f7521c9f1c3a831c162fe9a540df4b1/tensorflow/core/kernels/conv_ops_fused_impl.h#L216-L218).  I'll open an issue in the TF repo regarding this. So I agree that (3) might be the best option.

Lastly, wish you very happy new year!

Reply:
@mattdangerw 

Could you please take a look at this one? TIA

/cc @fchollet @qlzh727

Reply:
Still here on this! We think we have found a good workaround (option 3), forcing XLA compilation grouped convolutions. https://github.com/keras-team/keras/pull/15868

Once that lands (assuming that doesn't run into road blocks), we can submit this without modifying the predict call in load weights tests.

Reply:
@mattdangerw 

Thanks a lot for this! Really appreciate it.